server/game/game.js

```diff
@@ -814,6 +814,7 @@ class Game extends EventEmitter {
     }
 
     continue() {
+        this.effectEngine.reapplyStateDependentEffects();
```

Comment:
I'm a bit surprised this was sufficient to fix your test case, but I'm not sure it will work in the general case. What's really needed is to recalculate between each step in the pipeline, included nested pipelines like the individual phases (whereas this just reapplies before the first pipeline step after pausing for user input). Not sure how much overhead that will incur.